File: AdventOfCode/AoC_2015/Day1.cs
namespace AoC_2015;

public class Day1(string Input) : Utilities.BaseDay(Input)
{
    public override string Part1()
    {
        var line = File.ReadLines(InputFile).First();
        return line.Select(c => c == '(' ? 1 : -1).Sum().ToString();
    }

    public override string Part2()
    {
        var line = File.ReadLines(InputFile).First();
        var counter = 0;
        for (int i = 0; i < line.Length; i++)
        {
            counter += line[i] == '(' ? 1 : -1;
            if (counter == -1)
            {
                return (i + 1).ToString();
            }
        }

        return "ERROR";
    }
}

File: AoC2015/day1.py
"""Day 1: Not Quite Lisp"""

from Utilities.aoc_day import AdventOfCodeDay


class Day1(AdventOfCodeDay):
    """Day 1: Not Quite Lisp"""

    def part1(self):
        with open(self.input_file, "r", encoding="utf-8") as file:
            lines = file.readlines()
            return sum(1 if c == "(" else -1 for c in lines[0])

    def part2(self):
        with open(self.input_file, "r", encoding="utf-8") as file:
            lines = file.readlines()
            floor = 0
            for i, c in enumerate(lines[0]):
                floor += 1 if c == "(" else -1
                if floor < 0:
                    return i + 1

        return None
